spec/spec_helper.rb

```diff
@@ -194,6 +194,40 @@ def use_headful_chrome? = ENV.fetch('HEADFUL_CHROME', nil).present?
     RSpec::Support::ObjectFormatter.default_instance.max_formatted_output_length = 2_000
   end
 
+  config.around(:each, type: :feature) do |example|
+    $test_log_string_io ||= StringIO.new
```

Comment:
Unfortunately, it seems that this needs to be a global variable. (Or, at least, I didn't want to invest in finding a way to make this work without using a global variable.) Otherwise, what seems to happen is that only the first spec that is executed is able to write ActionController logs to the StringIO. If we try to do this as a local variable, then the later examples create a new StringIO, but I think that the Puma test server remains booted with a reference to the StringIO of the first example, and so, Puma doesn't write anything to the local StringIO that we might try to use for all examples other than the first one executed.